### ISSUE TITLE
Google forbids Huawei devices from using GMS, so how to control the browser when AppAuth works on these devices?

### ISSUE THREAD

**Issue by ttljtw** [openid/AppAuth-Android#819]
### Checklist:

- [x] I am using the latest release
- [x] I searched for [existing GitHub issues](https://github.com/openid/AppAuth-Android/issues)
- [x] I read the [documentation](https://github.com/openid/AppAuth-Android/blob/master/README.md)
- [x] I verified the client configuration matches the information in the identity provider (or I am using dynamic client registration)
- [x] I am either using a custom URI scheme or `https` with [App Links](https://developer.android.com/training/app-links) for client redirect.
- [x] I can reproduce the issue in the demo app (optional)

### Configuration
- Version: 0.11.1
- Integration: native(Java)
- Identity provider: Google

### Issue Description

I use a new Huawei device for application adaptation. My application requires a specified browser. Generally, we specify the pre installed Chrome browser of Android. But as you know that Huawei devices do not have GMS and will not be pre installed with Chrome. It is pre installed with Huawei browser. You can't expect every user to install Chrome on their devices. Therefore, can the parameters of Huawei browser be added to the versionedbrowsermatcher, so that this project is more perfect and more convenient for everyone to use.

**Comment by agologan** [comment]
Not sure what you're asking here. `VersionedBrowserMatcher` is just used to control BrowserAllowList/DenyList.

Out of the box you shouldn't restrict the browser list for best compatibility as the browser selector will just match available browsers in the order returned by the system.

This lib is not Chrome-specific or require GMS, though if you depend on AppLinks best support is provided by Chrome or forks of it.

**Comment by ttljtw** [comment]
Yes, I don't think it's a bug.
However, as an open source project, should developers be reminded in the README that there is restriction on some browser, which will lead to the decline of some user experience. For example,
![image](https://user-images.githubusercontent.com/11484469/160545054-b1e84da6-6b02-4251-94b2-b989522d8405.png)

**Comment by agologan** [comment]
Not really, it's not like we mention anywhere you need Chrome, and in all fairness if you use a custom scheme based URI it should work perfectly in all browsers including just the default webviews some obscure Android installs come with.

**Comment by EmDee** [comment]
`AppAuth-Android` is using [Custom Tabs](https://developer.chrome.com/docs/android/custom-tabs/), which is also supported by the Huawei browser for [close to two years now](https://www.huaweiupdate.com/huawei-browser-getting-the-latest-version-10-1-1-301/). I haven't tested this myself, but it should be working fine for you?

**Comment by ttljtw** [comment]
Yes, I understand what you're saying, if developers don't use advanced configuration([Controlling which browser is used for authorization](https://github.com/openid/AppAuth-Android#controlling-which-browser-is-used-for-authorization)), they won't be affected.
So, I wanted to add a notice to the advanced configuration to be more friendly to developers like me who choose to use the advanced configuration, didn't I?

**Comment by ttljtw** [comment]
> Now I get where you are coming from.
> 
> I guess it wouldn't hurt anyone, if there's an additional notice, however, I wouldn't mention Huawei specifically, but only state that the defined browsers might not be available on all the devices.

So, can I submit a notice to remind other developers of this?

**Comment by EmDee** [comment]
> 
> So, can I submit a notice to remind other developers of this?

I'm just another fellow developer. I think you can set up a PR and the maintainers will decide, if it's worth merging.
